### ISSUE TITLE
Weed Crop Image Classification

### ISSUE THREAD

**Issue by abhisheks008** [abhisheks008/DL-Simplified#249]
### Deep Learning Simplified Repository (Proposing new issue)
:red_circle: **Project Title** : Weed Crop Image Classification</br>
:red_circle: **Aim** : Aim is to create a DL model which will identify the images from the dataset.</br>
:red_circle: **Dataset** : https://www.kaggle.com/datasets/vinayakshanawad/weedcrop-image-dataset</br>
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Email ID :
- Participant ID (if applicable):
- Approach for this Project :
- What is your participant role? (Mention the Open Source program)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by aarushiksk** [comment]
Full Name: Aarushi
GitHub Profile Link: https://github.com/aarushiksk
Email ID: aarushiksk@gmail.com
Approach for this project: **Using CNN as well as some traditional ML algorithms like SVMs to see which performs better.**
What is your participant role? Participant in SSOC

**Comment by abhisheks008** [comment]
Hi @aarushiksk nice to have you here. I like your approach for solving this issue. This issue will be assigned to you once the program starts officially. 

Till then, star the repository and explore more in open source.

**Comment by NirmalKAhirwar** [comment]
FullName: Nirmal Kumar Ahirwar
Github Profile Link: https://github.com/NirmalKAhirwar
Email ID: [ahirwarnirmal2017@gmail.com](mailto:ahirwarnirmal2017@gmail.com)

Approach for this Project : 
1] Perform the Exploratory Data Analysis on the dataset
2] I will perform some Data Augmentation on the Dataset using Albumentations python library .
3] For Classification I will create different deep learning model of such as YOLOv8, Detectron, CNN models like VGG.
and compare the performance of them based on Accuracy metics.

What is your participant role? -contributor at Social Summer of Code (SSOC), Season-2

**Comment by abhisheks008** [comment]
Issue assigned to @NirmalKAhirwar

**Comment by yashpadhye17** [comment]
Full name : Yash Nimish Padhye
GitHub Profile Link : https://github.com/yashpadhye17
Email ID : yash.padhye.aids.2020@vpkbiet.org
Approach for this Project :
1) Application of different CNN Models.
2)Comparison of those models on Scaling Factors.

What is your participant role? (Mention the Open Source program): SSOC Season 2 (Contributor)

Query- If selected, can I team up with one more person if selected and she is also part of SSOC S2 ?

**Comment by abhisheks008** [comment]
Issue assigned to you @yashpadhye17

**Comment by Aryan863** [comment]
Full name : Aryan Kashyap
GitHub Profile Link : https://github.com/Aryan863
Email ID : aryanvkashyap03@gmail.com

Approach for this Project :

1. data preprocessing and visualization- To understand more about the dataset

2. data augmentation - apply augmentation techniques like flipping, rotation, brightness,etc to make the model more robust.

3. Model creation - I will be using VGG19, Xception, EfficientNet_v2 and MobileNet_v3 to compare the models.

4. Regularization- if model is overfit then try to regularize it.

5. Hyperparameter tuning- Experiment with hyperparameters such as no. of epochs, optimizers, add some additional layers to achieve more accurate and precise model.


What is your participant role?: Social Winter of code(SWOC) (Contributor)

**Comment by abhisheks008** [comment]
Issue assigned to you @Aryan863
